### DISCUSSION TITLE
Is there a documentation for using Web3Modal in Solana?

### DISCUSSION THREAD

**Discussion by 0xBeycan** [WalletConnect/walletconnect-monorepo#4545] [Developer Support] [answered]
Is there a documentation for using Web3Modal in Solana?

**Comment by glitch-txs** [comment, marked as answer]
The SDK is in alpha, but can check here https://walletconnect-docs-git-feat-web3modal-solana-walletconnect1.vercel.app/web3modal/react/about?platform=solana

**Comment by 0xBeycan** [comment]
thanks @glitch-txs, how is going Tron integration? I remember you changed my issue for the integration process.

**Reply by glitch-txs** [reply]
We are planning to make a huge refactor for multi-chain support on the next major, so it's on the works. Currently the Solana package only works with Solana.

**Reply by 0xBeycan** [reply]
Can I join the team? :D no need salary.

I am currently works on this: https://github.com/MultipleChain/js

**Reply by glitch-txs** [reply]
Thanks! We currently have an open position for engineering, feel free to apply 
https://apply.workable.com/walletconnect/j/95662E11FE/

Besides that you can always contribute as the tools are open source :)

Feel free to check on the open issues and you can also tag the team to get them assigned 👍

**Reply by 0xBeycan** [reply]
@glitch-txs That's good but I am not a Senior FE developer and not a React lover. My priority stack is BE and I am a Vue lover :D that's why I just want join the team to improve myself. No need salary.

You are right, I can contribute to the repo. But I don't know life cycle. But I will work on the below problems today. WalletConnect and Solflare.

UPDATE: I will apply DevRel and General Application.

**Comment by 0xBeycan** [comment]
https://github.com/MultipleChain/js/blob/beycan/packages/networks/solana/src/browser/adapters/Web3Modal.ts

I tried to integrate Web3Modal with the above code structure. Phantom works fine. However...

- Solflare gives network mismatch error. But I default to the selected network at startup, which I also use switchNetwork.

https://github.com/MultipleChain/js/blob/e5ffc1dbed7abbaa29e76e2bbaa0971c6a641439/packages/networks/solana/src/browser/adapters/Web3Modal.ts#L63

- WalletConnect connects but does not react to any action. like signing and sending tx.

- Also I guess it's because of a problem on the TrustWallet side or I'm not sure. But this package that was working before is not working now. It rejects the network.

https://github.com/MultipleChain/solana-walletconnect

If you have a chance to test it, I would appreciate it. Just download the packages and say "npm run dev" for the test environment. Everything is ready, you can just connect Web3Modal and test the actions.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/50718965/0f8f1a20-1d18-484f-8a3f-fe3d71806a51)

**Reply by 0xBeycan** [reply]
@glitch-txs can you confirm me?

**Reply by 0xBeycan** [reply]
Hey @glitch-txs  I noticed that WalletConnect does not work in stable packages that worked before. Web3Modal is fine. But after connecting with WalletConnect, there is no response back from the transactions sent. What is the possibility that this is only related to TrustWallet?

**Reply by 0xBeycan** [reply]
I just tested it by connecting to the mobile MetaMask and it was fine. I think this is a Web3Wallet issue related to @trustwallet. @rkokhatskyi

**Comment by 0xBeycan** [comment]
@glitch-txs I created a PR to fix Solflare network mismatch problem: https://github.com/WalletConnect/web3modal/pull/2298
